GOVERNANCE.md

```diff
@@ -0,0 +1,40 @@
+# API Improvement Proposals: Project Charter
```

Comment:
Please make sure this auto-formats with prettier to match the rest of the repository. (Additionally we should add CI to block if that isn't done).